docs/guides/basics/authentication/aws.md

```diff
@@ -0,0 +1,124 @@
+---
+nav: AWS
+description: Configure authentication for an Amazon Web Services (AWS) API.
+---
+
+# Authenticating using Amazon Web Services
+
+APIs that are part of Amazon Web Services (AWS) use a proprietary authentication scheme known as [AWS Signature Version 4][awsdocs_sv4]. Coda provides built-in support for this type of authentication, and to set it up you only need to write a few lines of simple code. You must specify which AWS service you are connecting to, and each Pack can only connect to a single service.
+
+The SDK supports two different ways of authenticating with an AWS service: access keys and by assuming a role.
+
+
+## Access key
+
+The simplest method of authentication requires that the Coda user provides an access key and secret associated with an AWS user. Coda will use these values to sign outgoing requests. Simply set the authentication type to [`AWSAccessKey`][sdk_awsaccesskey] and specify the AWS service you are connecting to.
+
+```ts
+pack.setUserAuthentication({
+  type: coda.AuthenticationType.AWSAccessKey,
+  service: "s3",
+});
+```
+
+When authenticating, the user will be prompted to enter their access key and secret.
+
+<img src="../../../../images/auth_aws_access_key.png" srcset="../../../../images/auth_aws_access_key_2x.png 2x" class="screenshot" alt="Prompting the user for their access key and secret">
+
+You can learn more about how to generate access keys in the [AWS documentation][awsdocs_accesskey].
+
+[View Sample Code][sample_aws]{ .md-button }
+
+
+## Assume role
+
+A more secure way to authenticate is by using a the [`AssumeRole` feature][awsdocs_assumerole] of the Security Token Service (STS). Trust is first established between the AWS user running Packs code and a role you created in your AWS account. When the Pack is run, temporary credentials are generated and used to sign outgoing requests.
+
+To use this form of authentication, simply set the authentication type to [`AWSAssumeRole`][sdk_awsassumerole] and specify the AWS service you are connecting to.
+
+```ts
+pack.setUserAuthentication({
+  type: coda.AuthenticationType.AWSAccessKey,
+  service: "s3",
+});
+```
+
+When authenticating, the user will be prompted to enter the Amazon Resource Name (ARN) of the role to assume, as well as an associated external ID.
+
+<img src="../../../../images/auth_aws_assume_role.png" srcset="../../../../images/auth_aws_assume_role_2x.png 2x" class="screenshot" alt="Prompting the user for their role and external ID">
+
+You can learn more about how to create roles in the [AWS documentation][awsdocs_createrole].
+
+
+### Trust policy
```

Comment:
Note: this needs to be setup by the pack user, not the pack author.

Reply:
Good catch, done.